addons/uom/views/uom_uom_views.xml

```diff
@@ -20,33 +20,12 @@
                 <sheet>
                     <group>
                          <group name="uom_details">
-                             <field name="name"/>
-                             <field name="category_id"/>
-                             <field name="uom_type" readonly="1"/>
-                             <label for="factor"
-                                invisible="uom_type != 'smaller'"/>
-                            <div invisible="uom_type != 'smaller'">
-                                <field name="factor"
-                                    digits="[42,5]"
-                                    readonly="uom_type == 'bigger'"/>
-                                <span class="opacity-50 oe_inline">
-                                    e.g: 1*(reference unit)=ratio*(this unit)
-                                </span>
+                            <field name="name" readonly="(context.get('product_id') or context.get('product_ids')) and id"/>
```

Comment:
What's the point of these weird `readonly` conditions?

Reply:
That's to make those fields readonly when the UoM form is opened from the many2many tags in the product form.
![image](https://github.com/user-attachments/assets/7b8353c4-eef6-49af-b004-076f951f7cff)

Reply:
I wonder how useful it is, since clicking on that button will make them editable anyway:
![image](https://github.com/user-attachments/assets/f13a2bb6-2e43-4656-9183-33bfa65d9692)